Mathlib/Analysis/SpecialFunctions/MulExpNegMulSq.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2024 Jakob Stiefel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jakob Stiefel
+-/
+import Mathlib.Analysis.SpecialFunctions.ExpDeriv
+import Mathlib.Analysis.SpecialFunctions.Log.Basic
+import Mathlib.Topology.ContinuousMap.Bounded.Basic
+
+/-!
+# Definition of `mulExpNegMulSq` and properties
+
+`mulExpNegMulSq` is the mapping `fun (ε : ℝ) (x : ℝ) => x * Real.exp (- (ε * x * x))`. By
+composition, it can be used to transform a function `g : E → ℝ` into a bounded function
+`mulExpNegMulSq ε ∘ g : E → ℝ = fun x => g x * Real.exp (-ε * g x * g x)` with useful
+boundedness and convergence properties.
+
+## Main Properties
+
+- `abs_mulExpNegMulSq_le`: For fixed `ε > 0`, the mapping `x ↦ mulExpNegMulSq ε x` is
+  bounded by `Real.sqrt ε⁻¹`;
+- `tendsto_mulExpNegMulSq`: For fixed `x : ℝ`, the mapping `mulExpNegMulSq ε x`
+  converges pointwise to `x` as `ε → 0`;
+- `lipschitzWith_one_mulExpNegMulSq`: For fixed `ε > 0`, the mapping `mulExpNegMulSq ε` is
+  Lipschitz with constant `1`;
+- `abs_mulExpNegMulSq_comp_le_norm`: For a fixed bounded continuous function `g`, the mapping
+  `mulExpNegMulSq ε ∘ g` is bounded by `norm g`, uniformly in `ε ≥ 0`;
+-/
+
+open NNReal ENNReal BoundedContinuousFunction Filter
+
+open scoped Topology
+
+namespace Real
+
+/-! ### Definition and properties of `fun x => x * Real.exp (- (ε * x * x))`-/
+
+/--
+Mapping `fun ε x => x * Real.exp (- (ε * x * x))`. By composition, it can be used to transform
+functions into bounded functions.
+-/
+noncomputable
+def mulExpNegMulSq (ε x : ℝ) := x * exp (- (ε * x * x))
+
+theorem mulExpNegSq_apply (ε x : ℝ) : mulExpNegMulSq ε x = x * exp (- (ε * x * x)) := rfl
+
+theorem neg_mulExpNegMulSq_neg (ε x : ℝ) : - mulExpNegMulSq ε (-x) = mulExpNegMulSq ε x := by
+  simp [mulExpNegMulSq]
+
+theorem mulExpNegMulSq_one_le_one (x : ℝ) : mulExpNegMulSq 1 x ≤ 1 := by
+  simp [mulExpNegMulSq]
+  rw [← le_div_iff₀ (exp_pos (-(x * x))), exp_neg, div_inv_eq_mul, one_mul]
+  apply le_trans _ (add_one_le_exp (x * x))
+  nlinarith
+
+theorem neg_one_le_mulExpNegMulSq_one (x : ℝ) : -1 ≤ mulExpNegMulSq 1 x := by
+  rw [← neg_mulExpNegMulSq_neg, neg_le_neg_iff]
+  exact mulExpNegMulSq_one_le_one (-x)
+
+theorem abs_mulExpNegMulSq_one_le_one (x : ℝ) : |mulExpNegMulSq 1 x| ≤ 1 :=
+  abs_le.mpr ⟨neg_one_le_mulExpNegMulSq_one x, mulExpNegMulSq_one_le_one x⟩
+
+variable {ε : ℝ}
+
+@[continuity, fun_prop]
+theorem continuous_mulExpNegMulSq : Continuous (fun x => mulExpNegMulSq ε x) :=
+  Continuous.mul continuous_id (by fun_prop)
```

Comment:
This is one place where having the compositional lemma you wrote earlier is useful, and where I should not have told you to remove it:
```suggestion
@[continuity, fun_prop]
theorem continuous_mulExpNegMulSq : Continuous (mulExpNegMulSq ε) :=
  Continuous.mul continuous_id (by fun_prop)
 
 @[continuity, fun_prop]
theorem _root_.Continuous.mulExpNegMulSq {α : Type*} [TopologicalSpace α] {f : α → ℝ} (hf : Continuous f) :
    Continuous (fun x => mulExpNegMulSq ε (f x)) :=
  continuous_mulExpNegMulSq.comp hf
```
That second lemma is an immediate consequence of the first but it allows you to write `hf.mulExpNegMulSq` for a short continuity proof.